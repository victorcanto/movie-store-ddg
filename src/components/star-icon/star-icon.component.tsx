import * as React from 'react';
import { Star as MuiStarIcon } from '@mui/icons-material';
import { Badge, Tooltip } from '@mui/material';

export function StarIcon() {
	return (
		<Tooltip title='NÃºmero de votos' data-testid='star-icon'>
			<MuiStarIcon
				data-testid='star-icon'
				sx={{ fill: 'yellow' }}
				fontSize='large'
			/>
		</Tooltip>
	);
}
