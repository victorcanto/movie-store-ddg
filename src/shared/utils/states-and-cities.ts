import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
