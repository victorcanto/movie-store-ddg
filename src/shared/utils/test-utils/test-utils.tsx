import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
