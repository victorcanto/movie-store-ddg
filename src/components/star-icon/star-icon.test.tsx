import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { StarIcon } from './star-icon.component';

test('snapshot of the base state', () => {
	const { asFragment } = render(<StarIcon />);
	expect(asFragment()).toMatchSnapshot();
});

test('should have in the document', () => {
	const { getByTestId } = render(<StarIcon />);
	expect(getByTestId('star-icon')).toBeInTheDocument();
});
