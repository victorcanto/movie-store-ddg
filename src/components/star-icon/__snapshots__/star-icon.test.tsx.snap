// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    aria-label="NÃºmero de votos"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-mey93b-MuiSvgIcon-root"
    data-mui-internal-clone-element="true"
    data-testid="star-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
    />
  </svg>
</DocumentFragment>
`;
